app/view/menuitem/LayerRefresh.js

```diff
@@ -0,0 +1,64 @@
+/**
+ * MenuItem to force redraw of a layer.
+ *
+ * @class CpsiMapview.view.menuitem.LayerRefresh
+ */
+Ext.define('CpsiMapview.view.menuitem.LayerRefresh', {
+    extend: 'Ext.menu.Item',
+    xtype: 'cmv_menuitem_layerrefresh',
+    requires: [],
+
+    /**
+     * The connected layer for this item.
+     *
+     * @cfg {ol.layer.Base}
+     */
+    layer: null,
+
+    /**
+     * Text shown in this MenuItem
+     * @cfg {String}
+     */
+    text: 'Refresh',
+
+
+    /**
+     * @private
+     */
+    initComponent: function () {
+        var me = this;
+        var allowRefresh = false;
+
+        if (me.layer) {
+            allowRefresh = me.layer.get('refreshLayerOption');
+        }
+
+        me.handler = me.handlerFunc;
+
+        me.callParent();
+
+        me.setHidden(!allowRefresh);
+    },
+
+    /**
+     * Executed when this menu item is clicked.
+     * Forces redraw of the connected layer.
+     */
+    handlerFunc: function () {
+        var me = this;
+        var source = me.layer.getSource();
+
+        // mostly WMS layers
+        if (source.updateParams) {
+            var params = source.getParams();
+            params.noCache = new Date().getMilliseconds();
+            source.updateParams(params);
+            source.refresh();
+        }
+
+        // mostly WFS layers
+        if (source instanceof ol.source.Vector) {
+            source.clear();
```

Comment:
shouldn't this be also a call of `refresh` instead of clear? Else the features will be gone

Reply:
Calling clear() forces the reload of the underlying data source. A refresh() does not query the source again. But you are right in case of having a vector layer with non-remote features they will be gone. Maybe we restrict this to WFS layers (however we detect them)?

Reply:
hmm ok, did not have that in mind, then clear should be ok.
For wfs detextion: i think every non-local vector layer should have an url or urlfunction defined, which is some kind of identifier for WFS...

Reply:
> For wfs detextion: i think every non-local vector layer should have an url or urlfunction (or loader) defined, which is some kind of identifier for WFS...

Unfortunately not. You can implement a custom `loader`. The URL is then undefined and a `loader` has no getter.  That's why I decided to introduce this custom property on the WFS layer.